ui/src/shared/components/TimeSeries.tsx

```diff
@@ -219,6 +226,10 @@ class TimeSeries extends Component<Props, State> {
         const extern = buildVarsOption([...vars, ...windowVars])
 
         event('runQuery', {context: 'TimeSeries'})
+        if (isCurrentPageDashboard) {
+          return getRunQueryResults(orgID, text, appState)
```

Comment:
Total suggestion: naming this function `checkCacheAndRunQuery` ?

Reply:
I'm gonna sound pedantic and say that might imply that we always `runQuery` which we don't in the case where the cache has a corresponding value. What about `getCachedResultsOrRunQuery`? A bit of a mouth full but definitely more explicit

Reply:
I love `getCachedResultsOrRunQuery`